File: examples/spring-io-2026/java/src/main/java/org/example/koog/spring/java/structs/AccountIssueSummary.java
package org.example.koog.spring.java.structs;

import ai.koog.agents.core.tools.annotations.LLMDescription;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

@LLMDescription("Full information about the user's issue with their bank account")
public record AccountIssueSummary(
        @JsonProperty("accountNumber")
        @LLMDescription("Account number of the user in the database")
        String accountNumber,

        @JsonProperty("username")
        @LLMDescription("Username of the account holder")
        String username,

        @JsonProperty("currentBalance")
        @LLMDescription("Current account balance in US dollars")
        Integer currentBalance,

        @JsonProperty("relatedTransactionId")
        @LLMDescription("ID of the transaction related to this issue, if applicable")
        String relatedTransactionId,

        @JsonProperty("disputeId")
        @LLMDescription("ID of the dispute if one was initiated for this issue")
        String disputeId,

        @JsonProperty("problem")
        @LLMDescription("What exactly is the user's issue with their account or transaction")
        String problem,

        @JsonProperty("resolved")
        @LLMDescription("Was the issue already resolved?")
        boolean resolved
) {
    @JsonCreator
    public AccountIssueSummary {
    }

    public static AccountIssueSummary some() {
        return new AccountIssueSummary("some-id", "Unknown", 0, null, null, "No issue", false);
    }
}

File: examples/spring-io-2026/kotlin/src/main/kotlin/org/example/koog/spring/kotlin/structs/AccountIssueSummary.kt
package org.example.koog.spring.kotlin.structs

import ai.koog.agents.core.tools.annotations.LLMDescription
import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import kotlinx.serialization.Serializable

@LLMDescription("Full information about the user's issue with their bank account")
@Serializable
data class AccountIssueSummary @JsonCreator constructor(
    @property:LLMDescription("Account number of the user in the database")
    val accountNumber: String,
    @property:LLMDescription("Username of the account holder")
    val username: String,
    @property:LLMDescription("Current account balance in US dollars")
    val currentBalance: Int,
    @property:LLMDescription("ID of the transaction related to this issue, if applicable")
    val relatedTransactionId: String,
    @property:LLMDescription("ID of the dispute if one was initiated for this issue")
    val disputeId: String,
    @property:LLMDescription("What exactly is the user's issue with their account or transaction")
    val problem: String,
    @property:LLMDescription("Was the issue already resolved?")
    val resolved: Boolean
)